LanguageFeatures/Augmentation-libraries/augmenting_constructors_A10_t09_lib.dart

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A non-redirecting generative constructor marked `augment` may:
+/// - Add or replace the body of the augmented constructor with a new body.
+/// ...
+///   - In the augmenting constructor's body, an `augmented()` call executes the
+///     augmented constructor's body in the same parameter scope that the
+///     augmenting body is executing in. The expression has type `void` and
+///     evaluates to `null`.
+///
+/// @description Checks that it is a compile-time error to use an `augmented()`
+/// in a non-augmenting constructor body.
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+augment library 'augmenting_constructors_A10_t09.dart';
+
+augment class C {
+  C() {
+    augmented();
```

Comment:
Depends on https://github.com/dart-lang/language/issues/4009, but I guess the consistent approach would be to follow the current spec text and then update all tests that are affected by a different spec wording, if the spec is updated on this topic.